src/vs/workbench/contrib/chat/browser/actions/quickQuestionActions/multipleByScrollQuickQuestionAction.ts

```diff
@@ -235,16 +246,48 @@ class QuickChat extends Disposable {
 	layout(): void {
 		if (this._currentParentElement) {
 			this.widget.layout(500, this._currentParentElement.offsetWidth);
+			// adjust height
+			if (this.chatViewOptions.renderInputOnTop) {
+				const offsetWidth = this._currentParentElement.offsetWidth;
+				// Needed to allow the widget to first be rendered so that the height of the container can be calculated
+				setTimeout0(() => {
+					const inputPartHeight = this.widget.inputPart.layout(500, offsetWidth);
+					const items = this.widget.viewModel?.getItems() ?? [];
+					switch (items.length) {
+						case 0:
+							this.widget.layout(inputPartHeight, offsetWidth);
```

Comment:
And I don't understand calling layout on the widget with inputPartHeight?

Reply:
If there are no messages, not even a welcome message, then the height should just be the height of the input part

Reply:
Ok, this seems backwards though, because the list should decide its height when it is empty